### DISCUSSION TITLE
Imitating Picasa's I feel Lucky

### DISCUSSION THREAD

**Discussion by Build-0-Matic** [ImageMagick/ImageMagick#6192] [Help] [answered]
### ImageMagick version

7.1.0-43

### Operating system, version and so on

Windows 10 64 bit

### Description

I have a question to which I have absolutely no idea as to where to start.
Anybody remembers Google Picassa? The image editor had an "I'm feeling lucky" button that would correct exposure and improve the color balance automatically.

I'm wondering if anybody here has an idea of how I could do the same in a Windows batch script?
Any idea is definitely welcome.

**Comment by fmw42** [comment]
If https://photo.stackexchange.com/questions/29536/what-does-the-im-feeling-lucky-function-in-picasa-3-do is true, then it is simply Imagemagick's `-auto-level`.  But you may prefer to balance each channel separately, if so, use `-channel rgb -auto-level`

**Reply by Build-0-Matic** [reply]
If that's the case, it's going to be simpler than I thought.
I just asked Chat GPT (yeah, I know, it spits out junk 90% of the time) and it says that

>  When clicked, the button would automatically apply various image enhancements, such as auto-contrast, auto-color, and auto-sharpen, to the selected photo.

But then it gave me functions that are not in the wiki.
After doing a bit of arm twisting, it told me to use
`-contrast-stretch 0 -normalize -unsharp 0x1`

Does that make any sense?

**Reply by fmw42** [reply]
-normalize is the same as contrast-stretch 2x1%. So no point in doing both.  -contrast-stretch 0 is essentially the same as -auto-level.  I would suggest just doing -auto-level -unsharp ...

If you do -normalize, you will clip pixel values at the top and bottom of the histogram. That will make bright values into white and dark values into black. You may not want to lose those intensities from your image.

**Reply by Build-0-Matic** [reply]
I like keeping things simple.
Thanks!

**Comment by fmw42** [comment]
Here are some examples.

Input:

![picassa_before](https://user-images.githubusercontent.com/13458218/227735722-aecb6547-fd58-4298-919f-45901a504a5f.jpg)

Picassa Result:

![picassa_after](https://user-images.githubusercontent.com/13458218/227735733-bb027837-a81c-4d6a-93bd-f4c90017dd42.jpg)


Auto-level:

`magick picassa_before.jpg -auto-level picass_before_autolevel.jpg`

![picass_before_autolevel](https://user-images.githubusercontent.com/13458218/227735765-71717ef9-6d8e-4395-afa9-49b5f5d4a074.jpg)


Auto-level each channel:

`magick picassa_before.jpg -channel rgb -auto-level picass_before_rgb_autolevel.jpg`

![picass_before_rgb_autolevel](https://user-images.githubusercontent.com/13458218/227735784-2a218890-a994-4eb1-907f-5c411f7e76b7.jpg)


Normalize:

`magick picassa_before.jpg -normalize picassa_before_normalize.jpg`

![picassa_before_normalize](https://user-images.githubusercontent.com/13458218/227735808-776b4431-5a7a-4526-b16a-94d35acb79fb.jpg)

Normalize each channel:

`magick picassa_before.jpg -channel rgb -normalize picassa_before_rgb_normalize.jpg`

![picassa_before_rgb_normalize](https://user-images.githubusercontent.com/13458218/227736215-9c4bc715-0b62-4eff-9338-d49a07a09817.jpg)


Normalize blend with and without channel processing:

```
magick picassa_before.jpg -write mpr:img +delete \
\( mpr:img -normalize \) \
\( mpr:img -channel rgb -normalize +channel \) \
-evaluate-sequence mean \
picass_before_normalize_rgb_blend.jpg

```

![picass_before_normalize_rgb_blend](https://user-images.githubusercontent.com/13458218/227736533-06b747aa-bd6b-4055-a195-44d02168a86d.jpg)


I have a script called vividize:

vividize -a 50 picassa_before.jpg picassa_before_vividize50.jpg

![picassa_before_vividize50](https://user-images.githubusercontent.com/13458218/227735889-9848b5dc-d11c-43c2-886f-1da88e0bb825.jpg)

**So based upon this one example, it looks like the -normalize blend is closest to what was used.**

**Reply by Build-0-Matic** [reply]
That is very interesting.
My photographer eye has the feeling that auto-level each channel combined with a gamma adjustment would probably be the closest...

I've got some scans with really wonky colors (due to a long expired film being used).
I'll have to try that...

**Reply by Build-0-Matic** [reply]
You're right, Normalize seems to do a better job.

**Reply by fmw42** [reply]
See the normalize blend example

**Reply by Build-0-Matic** [reply]
wow, this is getting to be really close.

**Comment by fmw42** [comment, marked as answer]
Here is another variation that is close. 

`convert picassa_before.jpg -channel rgb -contrast-stretch 1x0.25% picass_before_cs_1x0p25.png`

![picass_before_cs_1x0p25](https://user-images.githubusercontent.com/13458218/228051856-d55e3bd8-7e87-4f92-b50b-c0a8c839c8dd.png)

**Reply by Build-0-Matic** [reply]
As far as I'm concerned, this is probably the closest one so far.
So I'm calling it a winner.
Many thanks to you 😊
